rtt/transports/corba/CorbaDispatcher.hpp

```diff
@@ -123,29 +124,30 @@ namespace RTT {
              * @param iface
              */
             static void Release(DataFlowInterface* iface) {
+                if (!mlock) {
+                    return;
+                }
+
+                os::MutexLock lock(*mlock);
                 DispatchMap::iterator result = DispatchI.find(iface);
                 if ( result != DispatchI.end() ) {
-                    os::MutexLock lock(*mlock);
                     delete result->second;
                     DispatchI.erase(result);
                 }
-                if ( DispatchI.empty() )
-                    delete mlock;
-                mlock = 0;
             }
 
             /**
              * May be called during program termination to clean up all resources.
              */
             static void ReleaseAll() {
+                os::MutexLock lock(*mlock);
```

Comment:
Just remembering to do the same in `Release`

Reply:
Done